tests/test_autonomous_tasks.py

```diff
@@ -127,9 +122,7 @@ async def test_scheduled_tasks_are_enqueued_server_side(
     task_run: TaskRun = await foo_task_with_result_storage.submit(42)
```

Comment:
we shouldn't have to await this but that's because the `_create_autonomous_task_run` thing has the `sync_compatible` decorator on it which doesn't know _both_ that this is an async context _and_ that this task is sync - have a fix coming [here](https://github.com/PrefectHQ/prefect/pull/11806)